src/drive/web/modules/viewer/FilesViewer.jsx

```diff
@@ -35,140 +41,174 @@ export const FilesViewerLoading = () => (
+        try {
+          const fileCount = filesQuery.count
+
+          const currentIndex = files.findIndex(f => f.id === fileId)
+
+          if (
+            files.length !== fileCount &&
+            files.length - currentIndex <= 5 &&
+            isMounted
+          ) {
+            await filesQuery.fetchMore()
+          }
+        } finally {
+          setFetchingMore(false)
+        }
+      }
 
-      if (files.length !== fileCount && files.length - currentIndex <= 5) {
-        await filesQuery.fetchMore()
+      fetchMoreIfNecessary()
+
+      return () => {
+        isMounted = false
       }
-    } finally {
-      this.fetchingMore = false
-    }
+    },
+    [fetchingMore, filesQuery.count, files.length, fileId]
+  )
+
+  const currentIndex = useMemo(() => getCurrentIndex(), [getCurrentIndex])
+  const hasCurrentIndex = useMemo(() => currentIndex != -1, [currentIndex])
+  const viewerFiles = useMemo(() => (hasCurrentIndex ? files : [currentFile]), [
+    hasCurrentIndex,
+    files,
+    currentFile
+  ])
+  const viewerIndex = useMemo(() => (hasCurrentIndex ? currentIndex : 0), [
+    hasCurrentIndex,
+    currentIndex
+  ])
+
+  // If we can't find the file, we fallback to the (potentially loading)
+  // direct stat made by the viewer
+  if (currentIndex === -1 && !currentFile) {
+    return <FilesViewerLoading />
   }
 
-  onClose = () => {
-    if (this.props.onClose) {
-      this.props.onClose()
+  return (
+    <RemoveScroll>
+      <Overlay>
+        <Viewer
+          files={viewerFiles}
+          currentIndex={viewerIndex}
+          onChangeRequest={handleOnChange}
+          onCloseRequest={handleOnClose}
+          renderFallbackExtraContent={file => <Fallback file={file} t={t} />}
+          onlyOfficeProps={{
+            isEnabled: isOnlyOfficeEnabled(),
+            opener: file => router.push(makeOnlyOfficeFileRoute(file, true))
+          }}
+          panelInfoProps={{
+            showPanel,
+            PanelContent
+          }}
+          footerProps={{
+            FooterContent
+          }}
+        />
+      </Overlay>
+    </RemoveScroll>
+  )
+}
+
+const MemoisedFilesViewer = React.memo(FilesViewer)
+
+const styleStatusBar = switcher => {
+  if (window.StatusBar && isIOSApp()) {
+    if (switcher) {
+      window.StatusBar.backgroundColorByHexString('#32363F')
+      window.StatusBar.styleLightContent()
+    } else {
+      window.StatusBar.backgroundColorByHexString('#FFFFFF')
+      window.StatusBar.styleDefault()
     }
   }
+}
 
-  onChange = nextFile => {
-    if (this.props.onChange) {
-      this.props.onChange(nextFile.id)
-    }
+// We still keep the class component to set the style before rendering
+class FilesViewerWithStatusBarStyle extends Component {
+  componentWillMount() {
+    styleStatusBar(true)
   }
 
-  getCurrentIndex() {
-    const { files, fileId } = this.props
-    return files.findIndex(f => f.id === fileId)
+  componentWillUnmount() {
+    styleStatusBar(false)
   }
```

Comment:
I think you can go for the `useEffect(() => (), [])` instead of keeping `willMount`. If you really want to keep the willMount, you can play with `ref`: 
```jsx 
const useComponentWillMount = (func) => {
    const willMount = useRef(true)
    if (willMount.current) func()
    willMount.current = false
}
```